component/phlare/scrape/scrape_loop.go

```diff
@@ -126,7 +130,7 @@ func (tg *scrapePool) stop() {
 	defer tg.mtx.Unlock()
 
 	for _, t := range tg.activeTargets {
-		t.stop()
+		t.stop(true)
```

Comment:
I kinda like the `wait` idea on the stop method, I might use it in the future. 

My question is, this is only called when shutting down the component (eg. removing it from the config), right? Is there a reason why we want to wait here eg. to avoid leaking any goroutines or something?

Reply:
My idea is that we might attempt to shutdown the component, because we restart the agent and we don;'t want to interrupt goroutines until they have properly finished their last scrape.

All of this is assuming flow wait for the component.Run to end when shutting down. When just removing the component, it's less important as the goroutine will not be forcefully exited by os.Exit.

Reply:
Yes, the Agent will wait for all the component's Run methods to exit when shutting down eg. when an interrupt signal is received.

Could we see the same long waiting times when exiting as we did for updates? If so, I'd be a little worried about scheduling in Kubernetes being disrupted.

Reply:
Yes we probably need to stop in parallel to avoid long time instead of sequential. Should this be part of the current PR ?

Reply:
That's a good suggestion @tpaschalis, but can we push that off to another PR? I'd like for this to be in so we can stop the bleeding

Reply:
Yeah, let's go ahead and merge this!